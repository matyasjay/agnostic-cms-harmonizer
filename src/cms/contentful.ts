import {
  CreateClientParams as ContentfulClientParams,
  createClient,
} from 'contentful';

import { AgnosticCMSHarmonizerClient } from '..';
import { CmsClientName } from '../@types/client';

export class Contentful extends AgnosticCMSHarmonizerClient {
  constructor(clientParams: ContentfulClientParams) {
    super(CmsClientName.contentful, createClient, clientParams);
  }
}
