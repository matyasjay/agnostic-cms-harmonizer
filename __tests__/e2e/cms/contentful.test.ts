import { config as configDotenv } from 'dotenv';
configDotenv();


import { Contentful } from '../../../src/cms/contentful';

describe('contentful', () => {
  describe('should not fail', () => {
    it('retrieve content', async () => {
      const contentful = new Contentful({
        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN as string,
        environment: process.env.CONTENTFUL_ENVIRONMENT as string,
        space: process.env.CONTENTFUL_SPACE as string,
      });
      await contentful.initialize();
      const harmonizedData = await contentful.getEntry(
        process.env.CONTENTFUL_ENTRY as string,
      );
      expect(harmonizedData).toMatchSnapshot();
    });
  });
});
