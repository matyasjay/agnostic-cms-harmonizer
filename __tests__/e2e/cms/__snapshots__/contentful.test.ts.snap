// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentful should not fail retrieve content in default locale (en-US) 1`] = `
{
  "data": {
    "booleanInputFieldDefaultTrue": true,
    "jsonInputField": {
      "objectKey": "objectValue",
    },
    "questions": [
      {
        "configuration": {
          "identifier": "First name question configuration",
          "required": true,
        },
        "identifier": "First name question",
        "label": "First name",
        "placeholder": "Please enter your name",
      },
    ],
    "simpleShortInputTextAreaOne": "Amidst the chaos of dreams, a whisper of serenity finds its way.",
    "singleMediaInputField": "https://images.ctfassets.net/e5vout4n90kx/36j3phOyWhdSaqhRKeLauu/5bcf8eb2d10fdfb9d5af61f501d422ae/banner.jpg",
  },
}
`;

exports[`contentful should not fail retrieve content in specific locale \`es\` 1`] = `
{
  "data": {
    "booleanInputFieldDefaultTrue": true,
    "jsonInputField": {
      "objectKey": "objectValue",
    },
    "questions": [
      {
        "configuration": {
          "identifier": "First name question configuration",
          "required": true,
        },
        "identifier": "First name question",
        "label": "Nombre",
        "placeholder": "Por favor, introduzca su nombre",
      },
    ],
    "simpleShortInputTextAreaOne": "Amidst the chaos of dreams, a whisper of serenity finds its way.",
    "singleMediaInputField": "https:undefined",
  },
}
`;

exports[`contentful should not fail retrieve only one level of content in default locale (en-US) 1`] = `
{
  "data": {
    "booleanInputFieldDefaultTrue": true,
    "jsonInputField": {
      "objectKey": "objectValue",
    },
    "questions": [
      undefined,
    ],
    "simpleShortInputTextAreaOne": "Amidst the chaos of dreams, a whisper of serenity finds its way.",
    "singleMediaInputField": undefined,
  },
}
`;
